sips/sip-120.md

```diff
@@ -10,72 +10,70 @@ created: 2021-02-24
+Initially, TWAP prices will be sourced directly from Uniswap V3 pools with the expectation that the Synthetix system will only query large-liquidity markets such as WETH/WBTC and WETH/USDC. In the future, the oracle sources may be adjusted based on the assets enabled for atomic exchanges and their market conditions.
 
 ## Specification
 
 ### Overview
 
-To facilitate atomic exchanges, the `Synthetix` and `Exchanger` contracts will expose a new function, `exchangeAtomically()`. This new function will act in a similar manner to the current `Exchanger.exchange()` flow but with primary differences in:
+To facilitate atomic exchanges, the `Synthetix` and `Exchanger` contracts will expose a new function `exchangeAtomically()`. This new function will act in a similar manner to the current `Exchanger.exchange()` flow but with primary differences in:
 
 1. The execution price, detailed below
 1. Not having a fee reclamation window and therefore not minting any virtual synths
 1. Restrictions on source and destination synths, configurable by SCCPs
 
-Unlike `Exchanger.exchange()`, which relies solely on Chainlink oracles, the execution price for atomic exchanges is selected between the prices given by Chainlink oracles and Keep3r’s Uniswap and Sushiswap TWAP oracles. Three distinct prices are considered, `P_CLBUF`, `P_TWAP`, and `P_SPOT`, with the selected execution price being the one that outputs the minimum amount of destination synths:
+Unlike `Exchanger.exchange()`, which relies solely on Chainlink oracles, the execution price for atomic exchanges is selected between the prices given by Chainlink oracles and Uniswap V3 pools. Two distinct prices are considered, `P_CLBUF` and `P_TWAP`, with the selected execution price being the one that outputs the minimum amount of destination synths:
 
 - `P_CLBUF`: current Chainlink price, with a buffer of *N* bps applied against the trading direction (specified per-synth)
-- `P_TWAP`: minimum TWAP price between Uniswap and Sushiswap, over a window of *N* 30min periods
-- `P_SPOT`: minimum spot price between Uniswap and Sushiswap
+- `P_TWAP`: current Uniswap V3 TWAP price, over a window of *N* seconds
 
-`P_CLBUF` can be calculated internally within the current Synthetix system and `P_TWAP` and `P_SPOT` will be provided externally via [`Keep3rV4OracleUSD` (`0x471588ffe76c39815051f264d994cb45653cfd04`)](http://etherscan.io/address/0x471588ffe76c39815051f264d994cb45653cfd04#code).
+`P_CLBUF` can be calculated internally within the current Synthetix system and `P_TWAP` will be provided externally via [`UniswapV3CrossPoolOracle` (`0xeAAFD7547B781C60c71F0854a7dA2c1FF23a7dd0`)](https://etherscan.io/address/0xeaafd7547b781c60c71f0854a7da2c1ff23a7dd0#code).
```

Comment:
Uniswap are making a few tweaks to the lib this contract uses so I will update this address in the next few days, but I don't consider it a blocker for merging this PR.

Reply:
Done in cc5ac8e; barring anything major found later, this should be the last version of this oracle.

Backtesting may surface the need to incorporate other price sources, and those will be composed into a higher-level contract calling into this one. We can update this address later if so.